eras/shelley/impl/src/Cardano/Ledger/Shelley/UTxO.hs

```diff
@@ -163,11 +166,12 @@ shelleyProducedValue pp isRegPoolId txBody =
 getConsumedCoin ::
   ShelleyEraTxBody era =>
   PParams era ->
-  (StakeCredential (EraCrypto era) -> Maybe Coin) ->
+  (Credential 'Staking (EraCrypto era) -> Maybe Coin) ->
+  (Credential 'DRepRole (EraCrypto era) -> Maybe Coin) ->
```

Comment:
This function is not using the argument, thus it should not be even accepting that argument! This is always wrong to write functions like that. It is at the use site 
 that it should ignore the argument
```suggestion
```

It is in the instance declaration that you should ignore the argument:
```haskell
  getConsumedValue pp lookupRefund _ = getConsumedCoin pp lookupRefund
```

Reply:
Got it! :raised_hands:

Reply:
This is actually even worse. I had `getConsumedMaryValue` being used inside `conwayConsumedValue` and didn't realise the lookup function need not be provided there!